src/scheduler.rs

```diff
@@ -298,80 +309,123 @@ impl Scheduler {
         &self,
         original_req: UncheckedProposal,
     ) -> Result<String, SchedulerError> {
-        if original_req.is_output_substitution_disabled() {
-            return Err(SchedulerError::OutputSubstitutionDisabled);
-        }
+        let fallback_tx = original_req.get_transaction_to_schedule_broadcast();
+        log::debug!("original PSBT fallback tx: {:#?}", fallback_tx);
         let request = original_req
-            // This is interactive, NOT a Payment Processor, so we don't save original tx.
             // Humans can solve the failure case out of band by trying again.
-            .assume_interactive_receive_endpoint()
-            .assume_no_inputs_owned() // TODO Check
-            .assume_no_mixed_input_scripts() // This check is silly and could be ignored
-            .assume_no_inputs_seen_before(); // TODO
-
-        let mut original_psbt = request.psbt().clone();
-        info!("Received psbt: {:#?}", original_psbt);
+            .assume_interactive_receiver()
+            // TODO check
+            .check_inputs_not_owned(|_| false)?
+            .check_no_mixed_input_scripts()?
+            // TODO check, though this check is controversial since we're already breaking real "payjoin"
+            .check_no_inputs_seen_before(|_| false)?;
 
         // prepare proposal psbt (psbt, owned_vout, ScheduledPayJoin)
-        let (owned_vout, pj) =
-            self.find_matching_payjoin(&original_psbt).ok_or(SchedulerError::NoMatchingPayJoin)?;
-
-        // FIXME does this need to go before find_matching_payjoin(...) ?
-        // empty signatures because we won't broadcast the original psbt
-        original_psbt
-            .unsigned_tx
-            .input
-            .iter_mut()
-            .for_each(|txin| txin.script_sig = bitcoin::Script::default());
-
-        let our_output = &original_psbt.unsigned_tx.output[owned_vout];
-        if !pj.is_paying_us_enough(our_output) {
-            return Err(SchedulerError::OriginalPsbtInvalidAmount);
-        }
 
-        // initiate multiple `open_channel` requests and return the vector:
-        // Vec<(temporary_channel_id:, funding_txout:)>
-        let open_chan_results = pj.multi_open_channel(&self.lnd).await?;
-        let txouts_to_substitute: Vec<TxOut> =
-            open_chan_results.iter().map(|(_, txo)| txo.clone()).collect();
-        let temporary_chan_ids = open_chan_results.iter().map(|(id, _)| *id);
-
-        // TODO ensure privacy preserving txo ordering. should be responsibility of payjoin lib
+        log::debug!("find outputs that pay the receiver...");
+        let pj_by_script = self.pjs.lock().await;
+        log::debug!("pj_by_script: {:#?}", pj_by_script);
+        let mut payjoin_proposal = request.identify_receiver_outputs(|script| {
+            let addr =
+                Address::from_script(script, bitcoin::Network::Regtest).expect("script is valid");
```

Comment:
Regtest will not work in production. [ ] Read `lnd.get_info().chains.network` instead.